### ISSUE TITLE
Upscale Fails Immediately

### ISSUE THREAD

**Issue by JohnTravolski** [k4yt3x/video2x#109]
## Environment Information

|Module|Version|
|-|-| 
|`video2x`|video2x-2.7.1-win32-full.zip|
|`ffmpeg`|ffmpeg-20190330-5282cba-win64-static|
|`waifu2x-caffe`|ver 1.2.0.2|
|`waifu2x-converter-cpp`|Whatever was included in the zip|

## Symptom

When I attempt to run:
`./video2x -i testinput.mp4 -o out.mp4 -m gpu --width 3840 --height 2160`
I receive a "returned non-zero exit status" error.  The test input file can be found here:
https://drive.google.com/open?id=1kXXatUgsEq9aaWa9wkx-cuRxBBb0TcFu

Can somebody please help me understand what's going wrong?

## Screenshots or Error Messages

Please paste your error message in the code block down below:

```
PS E:\upscalingTesting\video2x> ./video2x -i testinput.mp4 -o out.mp4 -m gpu --width 3840 --height 2160
__      __  _       _                  ___   __   __
\ \    / / (_)     | |                |__ \  \ \ / /
 \ \  / /   _    __| |   ___    ___      ) |  \ V /
  \ \/ /   | |  / _` |  / _ \  / _ \    / /    > <
   \  /    | | | (_| | |  __/ | (_) |  / /_   / . \
    \/     |_|  \__,_|  \___|  \___/  |____| /_/ \_\

               Video2X Video Enlarger

                   Version 2.7.1

[!] WARNING: Specified cache directory C:\Users\UN2\AppData\Local\Temp\video2x does not exist
[?] USER: Create directory? [Y/n]: Y
[+] INFO: C:\Users\UN2\AppData\Local\Temp\video2x created
[+] INFO: Upscaling single video file: testinput.mp4
2019-06-23 16:44:46.608633 [+] INFO: Extracted frames are being saved to: C:\Users\UN2\AppData\Local\Temp\video2x\tmpc9y8gxhx
2019-06-23 16:44:46.609637 [+] INFO: Upscaled frames are being saved to: C:\Users\UN2\AppData\Local\Temp\video2x\tmpj6yj3n31
2019-06-23 16:44:46.610079 [+] INFO: Executing: ['E:\\upscalingTesting\\video2x\\dependencies\\ffmpeg-latest-win64-static\\bin\\ffmpeg.exe', '-hwaccel', 'auto', '-y', '-i', 'E:\\upscalingTesting\\video2x\\testinput.mp4', 'C:\\Users\\UN2\\AppData\\Local\\Temp\\video2x\\tmpc9y8gxhx\\extracted_%0d.png']
ffmpeg version N-93485-g5282cbae61 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 8.2.1 (GCC) 20190212
  configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libdav1d --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-amf --enable-ffnvcodec --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt
  libavutil      56. 26.100 / 56. 26.100
  libavcodec     58. 47.106 / 58. 47.106
  libavformat    58. 26.101 / 58. 26.101
  libavdevice    58.  7.100 / 58.  7.100
  libavfilter     7. 48.100 /  7. 48.100
  libswscale      5.  4.100 /  5.  4.100
  libswresample   3.  4.100 /  3.  4.100
  libpostproc    55.  4.100 / 55.  4.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'E:\upscalingTesting\video2x\testinput.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp42mp41
    creation_time   : 2019-06-23T21:40:22.000000Z
  Duration: 00:00:01.56, start: 0.000000, bitrate: 28702 kb/s
    Stream #0:0(eng): Video: h264 (Main) (avc1 / 0x31637661), yuv420p(tv, bt709), 1920x1080, 29049 kb/s, 59.94 fps, 59.94 tbr, 60k tbn, 119.88 tbc (default)
    Metadata:
      creation_time   : 2019-06-23T21:40:22.000000Z
      handler_name    : ?Mainconcept Video Media Handler
      encoder         : AVC Coding
    Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 317 kb/s (default)
    Metadata:
      creation_time   : 2019-06-23T21:40:22.000000Z
      handler_name    : #Mainconcept MP4 Sound Media Handler
[h264 @ 0000017106f6e500] Using auto hwaccel type dxva2 with new default device.
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> png (native))
Press [q] to stop, [?] for help
Output #0, image2, to 'C:\Users\UN2\AppData\Local\Temp\video2x\tmpc9y8gxhx\extracted_%0d.png':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp42mp41
    encoder         : Lavf58.26.101
    Stream #0:0(eng): Video: png, rgb24, 1920x1080, q=2-31, 200 kb/s, 59.94 fps, 59.94 tbn, 59.94 tbc (default)
    Metadata:
      creation_time   : 2019-06-23T21:40:22.000000Z
      handler_name    : ?Mainconcept Video Media Handler
      encoder         : Lavc58.47.106 png
frame=   91 fps= 11 q=-0.0 Lsize=N/A time=00:00:01.51 bitrate=N/A speed=0.181x
video:282357kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
[+] INFO: Reading video information
2019-06-23 16:44:55.332076 [+] INFO: Executing: E:\upscalingTesting\video2x\dependencies\ffmpeg-latest-win64-static\bin\ffprobe.exe -v quiet -print_format json -show_format -show_streams -i E:\upscalingTesting\video2x\testinput.mp4
[+] INFO: Framerate: 59.94005994005994
[+] INFO: Starting to upscale extracted images
2019-06-23 16:44:55.588932 [+] INFO: Starting upscaled image cleaner
2019-06-23 16:44:55.590445 [+] INFO: [upscaler] Thread 0 started
2019-06-23 16:44:55.591439 [+] INFO: Executing: ['E:\\upscalingTesting\\video2x\\dependencies\\waifu2x-caffe\\waifu2x-caffe-cui.exe', '--output_extention', 'png', '--mode', 'noise_scale', '--scale_width', '3840', '--scale_height', '2160', '--noise_level', '3', '--process', 'gpu', '--crop_size', '128', '--output_quality', '-1', '--output_depth', '8', '--batch_size', '1', '--gpu', '0', '--tta', '0', '--input_path', 'C:\\Users\\UN2\\AppData\\Local\\Temp\\video2x\\tmpc9y8gxhx\\0', '--output_path', 'C:\\Users\\UN2\\AppData\\Local\\Temp\\video2x\\tmpj6yj3n31']
Upscaling Progress:   0%|                                                                                     | 0/91 [00:00<?, ?it/s]変換に失敗したファイルがあります
2019-06-23 16:44:55.634551 [+] INFO: Killing upscaled image cleaner
Upscaling Progress:   0%|                                                                                     | 0/91 [00:01<?, ?it/s][!] ERROR: An exception has occurred
Traceback (most recent call last):
  File "video2x.py", line 318, in <module>
  File "upscaler.py", line 295, in run
  File "upscaler.py", line 234, in _upscale_frames
  File "waifu2x_caffe.py", line 83, in upscale
  File "subprocess.py", line 487, in run
subprocess.CalledProcessError: Command '['E:\\upscalingTesting\\video2x\\dependencies\\waifu2x-caffe\\waifu2x-caffe-cui.exe', '--output_extention', 'png', '--mode', 'noise_scale', '--scale_width', '3840', '--scale_height', '2160', '--noise_level', '3', '--process', 'gpu', '--crop_size', '128', '--output_quality', '-1', '--output_depth', '8', '--batch_size', '1', '--gpu', '0', '--tta', '0', '--input_path', 'C:\\Users\\UN2\\AppData\\Local\\Temp\\video2x\\tmpc9y8gxhx\\0', '--output_path', 'C:\\Users\\UN2\\AppData\\Local\\Temp\\video2x\\tmpj6yj3n31']' returned non-zero exit status 1.
```

**Comment by k4yt3x** [comment]
> 変換に失敗したファイルがあります

That's all that the output said. I'm not sure what's actually going wrong.
Might be a driver issue?

**Comment by JohnTravolski** [comment]
> > 変換に失敗したファイルがあります
> 
> That's all that the output said. I'm not sure what's actually going wrong.
> Might be a driver issue?

How do you advise I troubleshoot this issue?  I'm not sure where to start, and I'd really like to use this tool.

**Comment by k4yt3x** [comment]
@JohnTravolski Do you have CUDA drivers (just regular Nvidia GPU drivers) installed?

**Comment by DJPlaya** [comment]
Ime having the same Issue on AMD, latest stable Driver, Win 10 1809 64.

**Comment by ZaifSenpai** [comment]
I can not run it with gpu either. I am using AMD Radeon RX 560.

I found this repo which works fine with my gpu:
[](https://github.com/WL-Amigo/waifu2x-converter-cpp)

**Comment by k4yt3x** [comment]
@ZaifSenpai That engine is supported by video2x. Just switch the driver using `-d waifu2x_converter`.

**Comment by JohnTravolski** [comment]
> @JohnTravolski Do you have CUDA drivers (just regular Nvidia GPU drivers) installed?

Apparently I just needed to update my Graphics Driver.  Before, I had
`417.35-notebook-win10-64bit-international-whql-rp`
installed (NVIDIA Geforce GTX 980M).  When I updated to:
`430.86-notebook-win10-64bit-international-whql`
the issue was fixed and I am able to successfully use the GPU to perform the upscaling with no issues.  I'm still not sure why that old driver wouldn't work, but at least the issue is resolved now.
